src/movies.js

```diff
@@ -1,22 +1,122 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+
+    let directors = moviesArray.map(function(movies) {
+        return movies.director;
+    });
+
+    return directors;
+    
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+
+    let spielberg = moviesArray.filter((director) => director.director === 'Steven Spielberg')
+    let howMany = spielberg.filter((drama) => drama.genre.includes('Drama')) 
```

Comment:
Here, you don't need to create two variables, you can use the && operator to combine the two filter conditions into a single line of code, which will filter the array for movies directed by Steven Spielberg and that are also of the genre 'Drama'.
```
let howMany = moviesArray.filter((director) => director.director === 'Steven Spielberg' && director.genre.includes('Drama'));
```